juniper_official/Predictive-Maintenance/check-optical-temperature-netconf-ml.rule

```diff
@@ -0,0 +1,211 @@
+/*
+ * Checks for anomalies in optic module temperature 
+ *  
+ */
+healthbot {
+    topic linecard.optical {
+        rule check-optical-temperature-netconf-ml {
+            /*
+             * Monitors anomalies in optic module temperature
+             *
+             * Use interface-name as key for rule.
+             */		
+            keys interface-name;
+            synopsis "Optical module temperature KPI";
+            description "Check if  optical temp is having  anomaly ";
+            /*
+             * Sensor configuration to get data from network devices.
+             */			
+            sensor temperature {
+                iAgent {
+                    file opticalTemperature.yml;
+                    table module_TemperatureTable;
+                    frequency 60s;
+                }
+            }
+            /*
+             * Fields defined with variables and constant values
+             */			
+            field interface-name {
+                sensor temperature {
+                    where "name =~ /{{if-name}}/";
+                    path name;
+                }
+                type string;
+                description "Name of the interface.";
+            }
+            field opt-temp {
+                sensor temperature {
+                    path module-temperature;
+                }
+                type integer;
+                description "Temperature of the Optical Module in degrees Centigrade.";
+            }
+            field opt-temp-anomaly {
+                formula {
+                    anomaly-detection {
+                        algorithm 3sigma;
+                        learning-period 7days;
+                        pattern-periodicity 1days;
+                        field-name "$opt-temp";
+                    }
+                }
+                type integer;
+                description "Anomaly of optic module temperature";
+            }
+            /*
+             * Anomaly detection logic.
+             */			
+            trigger optical-temperature-anomaly {
```

Comment:
This is a confusing KPI name.
When the KPI "optical-temperature-anomaly" is green, does it mean that there is an anomaly - as in it is unhealthy; and conversely that it is healthy when red?
Can we rename this to optical-temperature-dt or even just optical-temperature (unless there is a clash there)? 
The more I think about it, the KPI shouldn't even have DT in it.. ML techniques are used to colour the KPI yellow, while breaching static thresholds color them red.